File: Medium/rps.py
# could improve this
# https://edabit.com/challenge/sfqudQHQ3HPpd7dZb
def rps(p1, p2):
    if p1==p2:
        return "It's a draw"
    elif p1=="Rock" and p2=="Paper":
        return "The winner is p2"
    elif p1=="Scissors" and p2=="Rock":
        return "The winner is p2"
    elif p1=="Scissors" and p2=="Paper":
        return "The winner is p1"
    elif p1=="Paper" and p2=="Rock":
        return "The winner is p1"


print(rps("Rock", "Paper"), "The winner is p2")
print(rps("Scissors", "Rock"), "The winner is p2")
print(rps("Scissors", "Paper"), "The winner is p1")
print(rps("Paper", "Rock"), "The winner is p1")
print(rps("Paper", "Paper"), "It's a draw")
print(rps("Rock", "Rock"), "It's a draw")

File: javascript/Very_Easy/rps.js


const rps = (p1, p2) => {
    // const options = ['rock','scissors','paper'];
    // const result = options.indexOf(p2)-options.indexOf(p1);
    // const result = [options.indexOf(p1),options.indexOf(p2)];
    // console.log(result);
    // return result<0?'Player 1 won!':result==0?'Draw':'Player 2 won!'
    if(p1==p2) return 'Draw!';
    if(p1=='scissors'&& p2=='paper') return 'Player 1 won!';
    if(p1=='paper' && p2=='rock') return 'Player 1 won!';
    if(p1=='rock' && p2=='scissors') return 'Player 1 won!';
    return 'Player 2 won!'
};

console.log(rps('rock', 'rock'), 'Draw!');
console.log(rps('scissors', 'scissors'), 'Draw!');
console.log(rps('paper', 'paper'), 'Draw!');
console.log(rps('scissors', 'rock'),'Player 2 won!');
console.log(rps('paper', 'scissors'), 'Player 2 won!');
console.log(rps('rock', 'paper'), 'Player 2 won!');
console.log(rps('rock', 'scissors'), 'Player 1 won!');
console.log(rps('scissors', 'paper'), 'Player 1 won!');
console.log(rps('paper', 'rock'), 'Player 1 won!');